ReactClient/react-client/src/components/SignIn/SignIn.jsx

```diff
@@ -0,0 +1,63 @@
+import React from 'react';
+import { useFormik } from 'formik';
+import GridWrapper from '../common/GridWrapper/GridWrapper';
+import { stylesSignIn } from './styles';
+
+const validate = values => {
+    const errors = {};
+    if (!values.email) {
+        errors.email = 'Email is required';
+    } else if (!/^[A-Z0-9._%+-]+@[A-Z0-9.-]+\.[A-Z]{2,4}$/i.test(values.email)) {
+        errors.email = 'Invalid email address';
+    }
+    if (!values.password) {
+        errors.password = 'Password is required';
+    } else if (values.password.length <= 6 || values.password.length >= 12) {
+        errors.password = 'Password incorrect length(6 min and 12 max)';
+    }
+
+    return errors;
+}
```

Comment:
Think about creating separate "helper" file that will contain different validators, so you can reuse them in the future.